dev/tools/pre-commit

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+# Copy to .git/hooks/ to use.
+
+set -e
+
+CODE=0
+git diff --cached --name-only -z | xargs -0 dev/tools/check-eof-newline.sh --fix || CODE=1
+
+if [ $CODE -ne 0 ]
+then
+    2>&1 echo "Some files had newline errors; they have been fixed in the working tree."
+    2>&1 echo "Make sure to add them before committing."
+    2>&1 echo "This may fix itself if you were using git commit -a, and you try again."
```

Comment:
Not fully clear what is meant here.

Reply:
>Not fully clear what is meant here.

The errors were fixed only in the worktree, so there was a missing `git add`.
Some thought had to be used to do that `git add` without adding other unstaged changes.
This is now fixed.